### ISSUE TITLE
Environment variable expansion

### ISSUE THREAD

**Issue by seanfarley** [xonsh/xonsh#1615]
It is unclear to me whether this was a design choice or an oversight but current versions of xonsh require `ls "$FOO/bar"` whereas bash expands `ls $FOO/bar` naturally.

**Comment by scopatz** [comment]
Thanks for reporting @seanfarley. I think this is a bug, and it is certainly annoying

**Comment by laerus** [comment]
`expandpath` works as expected
```
$ from xonsh.tools import expandpath
$ $FOO=10
$ expandpath('$FOO/spam')
'10/spam'
```

maybe this is a parsing bug?

**Comment by scopatz** [comment]
yeah, it is almoat certainly a parser bug where the parser is not generating and expandpath call in all cases.

**Comment by astronouth7303** [comment]
I didn't know this was a feature. I just figured `$SPAM` was treated as a variable name.

**Comment by scopatz** [comment]
A potential fix is now available in #1623
